bullet-journal/src/Components/Journal/VerticalLayout/DayView.js

```diff
@@ -1,68 +1,158 @@
-import React, {useState, Fragment} from 'react'
-import Bullet from './Bullet'
-
-function DayView(props) {
-
-    const mockBullets = [
-        {
-            date: '07/06/2020',
-            category: 'Work', 
-            text: 'Work on Bullet Journal'
-        },
-        {
-            date: '07/06/2020',
-            category: 'Work',
-            text: '3 easy coding problems'
-        },
-        {
-            date: '07/06/2020',
-            category: 'Casual',
-            text: 'Stream on Twitch'
-        },
-        {
-            date: '07/06/2020',
-            category: 'Casual',
-            text: 'Practice piano'
-        }
-    ]
-    
-    const [bullets, setBullets] = useState(mockBullets);
-    const [day, setDay] = useState({
-        name: 'Wednesday',
-        date: '07/01/2020'
-    })
+import React, {Fragment, useReducer, useRef} from 'react'
+import {v4 as uuid} from 'uuid'
+import Bullet from './Bullet';
+
+export const DayBulletsContext = React.createContext();
+
+const mockBullets = [
+    {
+        id: 1,
+        date: '07/06/2020',
+        category: 'Work', 
+        text: 'Work on Bullet Journal',
+        completed: false
+    },
+    {
+        id: 2,
+        date: '07/06/2020',
+        category: 'Work',
+        text: '3 easy coding problems',
+        completed: false
+    },
+    {
+        id: 3,
+        date: '07/06/2020',
+        category: 'Casual',
+        text: 'Stream on Twitch',
+        completed: false
+    },
+    {
+        id: 4,
+        date: '07/06/2020',
+        category: 'Casual',
+        text: 'Practice piano',
+        completed: false
+    }
+]
+
+const mockDay = {
+    name: 'Wednesday',
+    date: '07/01/2020'
+}
+
+const initialState = {
+    day: mockDay,
+    bullets: mockBullets
+}
 
-    const createBullet = () => {
+const bulletsReducer = (state, action) => {
+        
+    switch(action.type) {
+        case 'add' :
+            const b = {
+                id: uuid(),
+                date: state.day.date,
+                category: 'Work',
+                text: ''
+            }
+            return {...state, 
+                bullets: [...state.bullets, b]
+            }
 
+        case 'save' :
+            const i = state.bullets.findIndex(b => b.id === action.bullet.id)
+            const newBullets = [...state.bullets]
+            newBullets.splice(i, 1, action.bullet)
+            return {...state,bullets: newBullets}
+
+        case 'drag':
+            const curr = {...state.bullets[action.dragIndex]};
+            const copy = [...state.bullets]
+            copy.splice(action.dragIndex, 1)
+            copy.splice(action.enterIndex, 0, curr);
+            return {...state, bullets: copy}
+                            
+        case 'remove' :
+            return {...state, 
+                bullets: state.bullets.filter(e => e.id !== action.bullet.id)
+            }
+            
+        default:
+            return state
```

Comment:
Not reviewing the dragging feature for now; separate the dragging feature into its own branch for review